zetaclient/chains/bitcoin/observer/inbound.go

```diff
@@ -472,3 +472,173 @@ func GetBtcEvent(
 	}
 	return nil, nil
 }
+
+// GetBtcEventWithWitness either returns a valid BTCInboundEvent or nil.
```

Comment:
Can we move this function and below into a file like `witness.go`?

If we want to reorganized the code internally once the feature is completed, this will simplify in our case the process

Reply:
Done!